File: MyCSharpFunctions/Startup.cs
using Microsoft.Azure.Functions.Extensions.DependencyInjection;
using Microsoft.Extensions.DependencyInjection;
using System.Net.Http.Headers;

[assembly: FunctionsStartup(typeof(MyCSharpFunctions.Startup))]

namespace MyCSharpFunctions;

public sealed class Startup : FunctionsStartup
{
    public override void Configure(IFunctionsHostBuilder builder)
    {
        builder.Services.AddHttpClient("MyAuthorizedClient", client =>
        {
            client.DefaultRequestHeaders.Authorization =
                new AuthenticationHeaderValue("Bearer", "<token>");
        });
    }
}

File: MyFSharpFunctions/Startup.fs
namespace MyFSharpFunctions

open Microsoft.Azure.Functions.Extensions.DependencyInjection
open Microsoft.Extensions.DependencyInjection
open System.Net.Http.Headers

type Startup () = 
    inherit FunctionsStartup()  
        override this.Configure (builder: IFunctionsHostBuilder ) =  
            builder.Services.AddHttpClient("MyAuthorizedClient", (fun c -> 
                c.DefaultRequestHeaders.Authorization <- AuthenticationHeaderValue("Bearer", "<token>")
            )) |> ignore

[<assembly: FunctionsStartup(typeof<Startup>)>]
do ()